core/src/main/kotlin/org/evomaster/core/search/service/Archive.kt

```diff
@@ -616,6 +616,11 @@ class Archive<T> where T : Individual {
         return samplingCounter
     }
 
+    /**
+     * Expose the IdMapper used by the archive (read-only access pattern).
+     */
+    fun getIdMapper(): IdMapper = idMapper
```

Comment:
why this? `IdMapper` is an injectable service singleton. you can inject it directly where needed